src/requirements/requirement-exam-utils.ts

```diff
@@ -123,25 +135,75 @@ export default function getCourseEquivalentsFromUserExams(
   ].flat();
 }
 
-function toSubjects(data: ExamRequirements[]) {
-  const exams = data.map(({ name }) => name);
-  const subjects = [...new Set(exams)];
+const getExamFulfillment = (userExam: ExamTaken): ExamFulfillment | undefined => {
+  const exam = examData[userExam.examType][userExam.subject];
+  const fulfillment = exam.reduce((prev: ExamFulfillment | undefined, curr: ExamFulfillment) => {
+    // check if exam name matches and score is high enough
+    if (userExam.score >= curr.minimumScore) {
+      // update exam variable if this exam has a higher minimum score
+      if (!prev || prev.minimumScore < curr.minimumScore) {
+        return curr;
+      }
+    }
+    return prev;
+  }, undefined);
+  return fulfillment;
+};
+
+export const examsTakenToExamCourses = (exams: ExamsTaken): CourseTaken[] => {
```

Comment:
So does this mean that an exam can only have one course equivalent?

Reply:
One exam can have multiple course equivalents, but it will only be converted to a single exam course.

I think you might be conflating course equivalents and `courseTaken`s -- this does mean that in the new infra, an exam is represented by only one `courseTaken`.

Reply:
Gotcha, so this is only infra for getting up to the `CourseTaken` part, and later when you refactor the graph it'll do the calculations for course equivalents?

Reply:
And just to clarify - `courseTaken` is just the "middle ground"/connecting point between AP/IB and courses so that they're all represented by the same thing?

Reply:
Yeah, later it'll calculate the exam course's edges based on course equivalents.

`courseTaken` is the "courses" part. Ultimately, we want `courseTaken`s to be used in the requirements computation. Semester courses are converted to `courseTaken`s for the computation.

The middle ground would be course equivalents. In the old infra, we generated one `courseTaken` per course equivalent (that basically represented the same course as the course equivalent). In the new infra, we'll generate one `courseTaken` per exam (but still use the course equivalents to determine the edges / connected requirements).